lib/vsc/filesystem/oceanstor.py

```diff
@@ -310,7 +310,7 @@ def __init__(self, url, account, username, password):
 
         self.vsc = VSC()
         self.vscstorage = VscStorage()
-        self.host_institute = vsc.options.options.host_institute
+        self.host_institute = self.vsc.options.options.host_institute
```

Comment:
it's not enough. Add a call to `self.vsc.read_cfg()`

Reply:
not needed, that's done by default

Reply:
yes but `VscStorage()` explicitly calls it with `False` as arg...

Reply:
`VscStorage` is not playing any role here...
```
>>> from vsc.config.base import VSC
>>> vsc = VSC()
>>> vsc.options.options.host_institute
'brussel'
```

Reply:
it's a singleton class.

Trust me, I've tried this already on storctrl. I won't work :wink:

Reply:
ok, I see what you mean, this is a Python 2 thing. Fixed in https://github.com/vub-hpc/vsc-filesystem-oceanstor/pull/23/commits/7c39a993976aefb3fea207da35dce9beb9433964
```
Python 2.7.5 (default, Jun 28 2022, 15:30:04)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import config
>>> import oceanstor
>>> O = oceanstor.OceanStorOperations(config.OCEANSTOR_URL, config.OCEANSTOR_ACCOUNT, config.OCEANSTOR_USER, config.OCEANSTOR_PWD)
>>> O.host_institute
'brussel'
```